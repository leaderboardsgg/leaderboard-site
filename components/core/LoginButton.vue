<template>
  <BaseButtonLink
    :class-names="classNames"
    class="login-button"
    link-to="#"
  >
    Log In
  </BaseButtonLink>
</template>

<script>
export default {
  props: {
    classNames: {
      default: '',
      type: String
    }
  }
};
</script>

<style scoped>
  @layer components {
    .login-button {
      @apply mx-1 py-1 hover:bg-gray-100;
    }
  }
</style>
