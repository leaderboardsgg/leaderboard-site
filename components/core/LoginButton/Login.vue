<template>
  <BaseButtonLink
    :class-names="classNames + `
      py-2
      hover:bg-gray-100
    `"
    link-to="#"
  >
    Log In
  </BaseButtonLink>
</template>

<script>
export default {
  props: {
    classNames: {
      default: '',
      type: String
    }
  }
};
</script>
