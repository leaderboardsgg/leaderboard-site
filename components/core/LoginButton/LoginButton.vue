<template>
  <BaseButton class="login-button" v-bind="$attrs" v-on="$listeners">
    Log In
  </BaseButton>
</template>

<style scoped>
.login-button {
  @apply mx-1 py-1 hover:bg-gray-100;
}
</style>
