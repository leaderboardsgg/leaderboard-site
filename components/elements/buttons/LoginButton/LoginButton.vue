<script setup lang="ts">
import BaseButton from 'elements/buttons/BaseButton/BaseButton.vue'
</script>

<template>
  <BaseButton class="login-button" v-bind="$attrs">
    {{ $t('login') }}
  </BaseButton>
</template>

<style lang="postcss" scoped>
.login-button {
  @apply mx-1 py-1 hover:bg-gray-100;
}
</style>
