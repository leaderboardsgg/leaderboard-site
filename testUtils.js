import Vue from 'vue';
import { RouterLinkStub } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (Component, options) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
