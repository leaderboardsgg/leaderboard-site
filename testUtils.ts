import Vue, { ComponentOptions } from 'vue';
import { RouterLinkStub, VueClass } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render, RenderOptions } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component<
  { tag: string; to: string },
  unknown,
  unknown
>('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data?.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (
  Component: VueClass<Vue> | ComponentOptions<Vue>,
  options?: RenderOptions<Vue>,
) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
