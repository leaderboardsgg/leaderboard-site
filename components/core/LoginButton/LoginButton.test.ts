import { stubbedRender } from '@/testUtils';
import LoginButton from './LoginButton.vue';

describe('<LoginButton />', () => {
  it('should render without crashing', () => {
    const { unmount } = stubbedRender(LoginButton);

    unmount();
  });

  it('renders correctly', () => {
    const { container } = stubbedRender(LoginButton);

    expect(container.firstChild).toMatchSnapshot();
  });
});
