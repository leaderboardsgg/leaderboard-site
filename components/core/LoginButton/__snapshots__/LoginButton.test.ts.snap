// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> renders correctly 1`] = `
<a
  class="button-link login-button"
  href="#"
  to="#"
>
  
  Log In

</a>
`;
